core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt

```diff
@@ -0,0 +1,429 @@
+                            _uiState.value.personPicture
+                        )
+                    }
+                    if (passkeyCreated != null) {
+                        navigateToAppropriateScreen(savePerson)
+                    }
+                } else {
+                    val passwordVal = _uiState.value.password ?: return@launch
+                    accountManager.register(
+                        person = savePerson,
+                        password = passwordVal,
+                        learningSpaceUrl = serverUrl
+                    )
+                }
+
+
+                val personPictureVal = _uiState.value.personPicture
+                if (personPictureVal != null) {
+                    personPictureVal.personPictureUid = savePerson.personUid
+                    personPictureVal.personPictureLct = systemTimeInMillis()
+                    val personPictureUriVal = personPictureVal.personPictureUri
+
+                    enqueueSavePictureUseCase(
+                        entityUid = savePerson.personUid,
+                        tableId = PersonPicture.TABLE_ID,
+                        pictureUri = personPictureUriVal
+                    )
+
+                }
+
+                navigateToAppropriateScreen(savePerson)
+
+            } catch (e: Exception) {
+                if (e is IllegalStateException) {
+                    _uiState.update { prev ->
+                        prev.copy(usernameError = systemImpl.getString(MR.strings.person_exists))
+                    }
+                } else {
+                    snackDispatcher.showSnackBar(
+                        Snack(systemImpl.getString(MR.strings.login_network_error))
+                    )
+                }
+                return@launch
+            } finally {
+                loadingState = LoadingUiState.NOT_LOADING
+            }
+        }
+    }
+
+    private fun navigateToAppropriateScreen(savePerson: Person) {
+        if (_uiState.value.isTeacher) {
+            val goOptions = UstadMobileSystemCommon.UstadGoOptions(clearStack = true)
+            Napier.d { "AddSignUpPresenter: go to next destination: $nextDestination" }
+            navController.navigateToViewUri(
+                nextDestination.appendSelectedAccount(
+                    savePerson.personUid,
+                    LearningSpace(accountManager.activeLearningSpace.url)
+                ),
+                goOptions
+            )
+        }
+        if (_uiState.value.isParent) {
+            navController.navigate(AddChildProfileViewModel.DEST_NAME, emptyMap())
+        }
+    }
+
+    fun onClickOtherOption() {
+
+        navController.navigate(OtherSignUpOptionSelectionViewModel.DEST_NAME, emptyMap())
+
+    }
+
+    companion object {
+
+        const val STATE_KEY_PICTURE = "picState"
+
+        const val DEST_NAME = "SignUp"
+
+        const val ARG_DATE_OF_BIRTH = "DateOfBirth"
+
+        val REGISTRATION_ARGS_TO_PASS = listOf(
```

Comment:
This must not be copy/pasted. We should reference the registration_args_to_pass we already have. There should only be one - otehrwise if an argument is included in one, but not the other, that will create a problem. 

Key principle: single point of truth.